skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurationType.kt

```diff
@@ -39,5 +44,28 @@ class SkaffoldRunConfigurationType : ConfigurationType {
 
     override fun getDisplayName(): String = message("skaffold.run.config.general.name")
 
-    override fun getConfigurationFactories(): Array<ConfigurationFactory> = emptyArray()
+    override fun getConfigurationFactories(): Array<ConfigurationFactory> =
+        arrayOf(skaffoldSingleRunConfigurationFactory, skaffoldDevConfigurationFactory)
+}
+
+/**
+ * Factory for Skaffold single run ("skaffold run" style) configurations. See template configuration
+ * at [SkaffoldSingleRunConfiguration].
+ */
+class SkaffoldSingleRunConfigurationFactory(type: ConfigurationType) : ConfigurationFactory(type) {
+    override fun createTemplateConfiguration(project: Project): RunConfiguration =
+        SkaffoldSingleRunConfiguration(project, this, name)
+
+    override fun getName(): String = message("skaffold.run.config.single.run")
+}
+
+/**
+ * Factory for Skaffold dev (continuous) mode ("skaffold dev" style) configurations.
+ * See template configuration at [SkaffoldDevConfiguration].
+ */
+class SkaffoldDevConfigurationFactory(type: ConfigurationType) : ConfigurationFactory(type) {
+    override fun createTemplateConfiguration(project: Project): RunConfiguration =
```

Comment:
could you remind me the diff between createConfiguration and createTemplateConfiguration?

Reply:
I think management code calls `createTemplateConfiguration` to create a "template" instance and then clones it to create an actual configuration in default implementation of `createConfiguration`.

Reply:
sg